web/src/routes/(user)/albums/[albumId]/+page.svelte

```diff
@@ -384,176 +433,225 @@
-              disabled={album.assetCount == 0}
-              on:click={() => (viewMode = ViewMode.SELECT_USERS)}
-            >
-              Share
-            </Button>
-          {/if}
-        </svelte:fragment>
-      </ControlAppBar>
-    {/if}
+            {#if album.assetCount > 0}
+              <CircleIconButton title="Download" on:click={handleDownloadAlbum} icon={mdiFolderDownloadOutline} />
 
-    {#if viewMode === ViewMode.SELECT_ASSETS}
-      <ControlAppBar on:close-button-click={handleCloseSelectAssets}>
-        <svelte:fragment slot="leading">
-          <p class="text-lg dark:text-immich-dark-fg">
-            {#if $timelineSelected.size == 0}
-              Add to album
-            {:else}
-              {$timelineSelected.size.toLocaleString($locale)} selected
+              {#if isOwned}
+                <div use:clickOutside on:outclick={() => (viewMode = ViewMode.VIEW)}>
+                  <CircleIconButton title="Album options" on:click={handleOpenAlbumOptions} icon={mdiDotsVertical}>
+                    {#if viewMode === ViewMode.ALBUM_OPTIONS}
+                      <ContextMenu {...contextMenuPosition}>
+                        <MenuOption on:click={() => (viewMode = ViewMode.SELECT_THUMBNAIL)} text="Set album cover" />
+                      </ContextMenu>
+                    {/if}
+                  </CircleIconButton>
+                </div>
+              {/if}
             {/if}
-          </p>
-        </svelte:fragment>
-
-        <svelte:fragment slot="trailing">
-          <button
-            on:click={handleSelectFromComputer}
-            class="rounded-lg px-6 py-2 text-sm font-medium text-immich-primary transition-all hover:bg-immich-primary/10 dark:text-immich-dark-primary dark:hover:bg-immich-dark-primary/25"
-          >
-            Select from computer
-          </button>
-          <Button size="sm" rounded="lg" disabled={$timelineSelected.size === 0} on:click={handleAddAssets}>Done</Button
-          >
-        </svelte:fragment>
-      </ControlAppBar>
-    {/if}
 
-    {#if viewMode === ViewMode.SELECT_THUMBNAIL}
-      <ControlAppBar on:close-button-click={() => (viewMode = ViewMode.VIEW)}>
-        <svelte:fragment slot="leading">Select Album Cover</svelte:fragment>
-      </ControlAppBar>
+            {#if isCreatingSharedAlbum && album.sharedUsers.length === 0}
+              <Button
+                size="sm"
+                rounded="lg"
+                disabled={album.assetCount == 0}
+                on:click={() => (viewMode = ViewMode.SELECT_USERS)}
+              >
+                Share
+              </Button>
+            {/if}
+          </svelte:fragment>
+        </ControlAppBar>
+      {/if}
+
+      {#if viewMode === ViewMode.SELECT_ASSETS}
+        <ControlAppBar on:close-button-click={handleCloseSelectAssets}>
+          <svelte:fragment slot="leading">
+            <p class="text-lg dark:text-immich-dark-fg">
+              {#if $timelineSelected.size == 0}
```

Comment:
This should be `===`. Doesn't matter in that case because `$timelineSelected.size` cannot be undefined or otherwise falsy. However I think it is good practice to always check for the explicit type, if that's the intention of the expression.